src/Keboola/StorageApi/Client.php

```diff
@@ -1330,10 +1331,29 @@ public function uploadFile($filePath, FileUploadOptions $options)
                 $result['name']
             );
 
+            $cnt = 0;
             do {
                 try {
+                    $cnt++;
```

Comment:
čítač by měl bejt určitě v `catch`, ne v tomhle nekonečným wait loopu.

Reply:
neni to jedno?

Reply:
určitě ne. `$uploader->upload();` je jen wrapper pro `$uploader->promise()->wait();`, takže to lítá ve smyčce a čeká (nedokázal jsem dohledat, jak rychle), takže nepočítáš počet pokusů, ale počet iterací "asynchronního" waiteru. chceš přece spíš kontrolovat počet retryů na upload, ne?

Reply:
no praveze je to wrapper pro wait, takze ta smycka je uvnitr, ten do-while je tam jen jako retry, ne kvuli tomu uploadu, takze je to pocet pokusu

(disclaimer: samozrejme to logicky patri spis do toho catch, ale chci se ujistit, ze neco neprehlizim, protoze funkcne to proste musi byt stejny)

Reply:
Koukám, že do sliced uploadu jsem to implementoval stejně, tak bych to řešení 1:1 zkopíroval

https://github.com/keboola/storage-api-php-client/blob/23f82f325f26c53fb7cc4807015a431a943343fb/src/Keboola/StorageApi/Client.php#L1582-L1585